src/rgw/rgw_op.cc

```diff
@@ -3749,7 +3922,10 @@ int RGWDeleteObj::handle_slo_manifest(bufferlist& bl)
 
 int RGWDeleteObj::verify_permission()
 {
-  if (!verify_bucket_permission(s, RGW_PERM_WRITE)) {
+  if (!verify_object_permission(s,
+				s->object.instance.empty() ?
```

Comment:
Thank you. It was intentional, but obviously not the right thing to do. We need to check the object name against the bucket policy which verify_object_permissions doesn't. I'll open-code it. Thank you!